services/rsources/extension.go

```diff
@@ -0,0 +1,130 @@
+package rsources
+
```

Comment:
[minor] To keep things tight I would create a new folder / sub-package named `extensions` or `repos` / `repositories`.

Then `extension.go` and `multitenant.go` could be moved there.

Reply:
I propose to use this structure: `rsources/internal/extensions/extension.go`
We would need to export the `extension` interface and its producer methods, so let's use the `internal` keyword to highlight that it is not intended for actual use by third-parties.